File: Programmers/Lv.1/소수찾기.py
def is_prime(num):
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

def solution(n):
    answer = 0
    for i in range(2, n + 1):
        if is_prime(i):
            answer += 1
    return answer

# def solution(n):
#     num = set(range(2, n + 1))

#     for i in range(2, n + 1):
#         if i in num:
#             num -= set(range(2 * i, n + 1, i))
#     return len(num)

File: Programmers/Lv.2.Kotlin/소수찾기.kt
class Solution {
    lateinit var answer: MutableSet<Int>
    
    fun solution(numbers: String): Int {
        answer = mutableSetOf()
        getCombination(numbers, "")
        return answer.filter { isPrime(it) }.count()
    }
    
    fun getCombination(numbers: String, result: String) {
        if (result.isNotEmpty()) answer.add(result.toInt())
        if (numbers.isEmpty()) return
        
        numbers.forEachIndexed { index, c ->
            getCombination(numbers.removeRange(index..index), c.plus(result))
        }
    }
    
    fun isPrime(number: Int): Boolean {
        if (number == 1 || number == 0) {
            return false
        }
        
        for (i in (2..(number / 2))) {
            if (number % i == 0) {
                return false
            }
        }
        
        return true
    }
}